server/src/logic.js

```diff
@@ -338,23 +340,26 @@ export const getHandlers = ({ models, creators: Creators }) => {
           .then((result) => {
             const updateArgs = { ...args.response };
             delete updateArgs.id;
-            return event.getEventlocations({ where: { id: updateArgs.destination.id } })
-              .then((destination) => {
-                if (!destination) {
-                  return Promise.reject(Error('Unknown destination passed'));
-                }
-                delete updateArgs.destination;
-                updateArgs.eventlocationId = destination[0].id;
-                if (result.length > 0) {
-                  return result[0].update(updateArgs);
-                }
-
-                return Creators.eventResponse({
-                  event,
-                  user,
-                  ...updateArgs,
+            if (updateArgs.destination) {
+              return event.getEventlocations({ where: { id: updateArgs.destination.id } })
+                .then((destination) => {
+                  if (!destination) {
+                    return Promise.reject(Error('Unknown destination passed'));
+                  }
+                  delete updateArgs.destination;
+                  updateArgs.eventlocationId = destination[0].id;
+                  if (result.length > 0) {
+                    return result[0].update(updateArgs);
+                  }
+                  return Creators.eventResponse({
+                    event,
+                    user,
+                    destination: destination[0],
+                    ...updateArgs,
+                  });
                 });
-              });
+            }
+            return result[0].update(updateArgs);
```

Comment:
so if there is no `destination` we bypass the `EventLocations` lookup and always update.
What if there is no `destination` and there is no existing `eventResponse`? If you intentionally didn't want to handle that situation I think we need to check for it.. Keep in mind it also breaks the idempotency of this function though. :( (the idea is you call setResponse() regardless of whether a response exists already and it atomically replaces if one exists or creates a new one if it doesn't)

If you want to add the check and ensure it is clear we don't handle it before L362 add:
```
if (!destination) {
  return Promise.reject(Error('Destination must be provided to respond to a event that does not already have a response'));
}